Sources/NetworkProtection/NetworkProtectionDeviceManager.swift

```diff
@@ -46,7 +46,7 @@ public protocol NetworkProtectionDeviceManagement {
                                      includedRoutes: [IPAddressRange],
                                      excludedRoutes: [IPAddressRange],
                                      isKillSwitchEnabled: Bool,
-                                     regenerateKey: Bool) async throws -> (TunnelConfiguration, NetworkProtectionServerInfo)
+                                     keyPair: KeyPair) async throws -> (TunnelConfiguration, NetworkProtectionServerInfo)
```

Comment:
I'm moving out the key generation responsibility to the PacketTunnelProvider, because I want ANY failure that may happen during rekeying to result in the key NOT being updated.

Additionally, several other methods to handle rekeying logic are there, so this movement will make it easier to group all of these snippets into a class specifically designed to this end.

I actually started implementing said class in this PR but eventually decided it was too many changes for a single PR - this one should be about fixing rekeying specifically.